From 398badd497bf95bb20ae6ac262d45705c049dfe0 Mon Sep 17 00:00:00 2001
From: Wei-Luan Wang <weiluanwang@google.com>
Date: Thu, 26 Feb 2026 01:06:03 +0000
Subject: [PATCH] Revert "Reset the scheduler policy when the thread type is
 not kRealtimeAudio"

This reverts commit 727004f01904fb0cc9bd87a3ae78dbe47ae3ab87.
---
 base/threading/platform_thread_cros.cc | 23 ++++++++++-------------
 1 file changed, 10 insertions(+), 13 deletions(-)

diff --git a/base/threading/platform_thread_cros.cc b/base/threading/platform_thread_cros.cc
index 432f4721f6..85ad441e20 100644
--- a/base/threading/platform_thread_cros.cc
+++ b/base/threading/platform_thread_cros.cc
@@ -232,25 +232,22 @@ void SetThreadTypeOtherAttrs(ProcessId process_id,
   SetThreadLatencySensitivity(process_id, thread_id, thread_type);
 }
 
-// Set or reset the RT priority of a thread based on its type.
+// Set or reset the RT priority of a thread based on its type
+// and whether the process it is in is backgrounded.
 // Setting an RT task to CFS retains the task's nice value.
 void SetThreadRTPrioFromType(ProcessId process_id,
                              PlatformThreadId thread_id,
                              ThreadType thread_type) {
-  struct sched_param prio{};
-  int policy;
-  pid_t syscall_tid =
-      thread_id == PlatformThread::CurrentId() ? 0 : thread_id.raw();
-
-  if (thread_type == ThreadType::kRealtimeAudio) {
-    prio = PlatformThreadChromeOS::kRealTimeAudioPrio;
-    policy = SCHED_RR;
-  } else {
-    prio.sched_priority = 0;
-    policy = SCHED_OTHER;
+  // TODO(crbug.com/262267726): Fix the issue that when thread type changed
+  // from kRealtimeAudio to other, the scheduling policy is still SCHED_RR.
+  if (thread_type != ThreadType::kRealtimeAudio) {
+    return;
   }
 
-  if (sched_setscheduler(syscall_tid, policy, &prio) != 0) {
+  pid_t syscall_tid =
+      thread_id == PlatformThread::CurrentId() ? 0 : thread_id.raw();
+  if (sched_setscheduler(syscall_tid, SCHED_RR,
+                         &PlatformThreadChromeOS::kRealTimeAudioPrio) != 0) {
     DVPLOG(1) << "Failed to set policy/priority for thread " << thread_id;
   }
 }
-- 
2.53.0.414.gf7e9f6c205-goog

